### PR TITLE
Fix kfp template resource rendering when exporting

### PR DESCRIPTION
**Pull request by akchinSTC**
- Render resource requests in kfp template only when values exist
- Remove 'G' memory unit from template since notebook operation
will handle

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by akchinSTC** [comment]
Memory unit needed to be removed from template since it is handled by the notebook operator and template will now only populate resources with non zero values.
previous template :
image
updated template:
image

**Review by lresende — APPROVED** [review]

File: elyra/templates/kfp_template.jinja2 [hunk 1]
```diff
@@ -22,9 +22,15 @@ def create_pipeline():
...
                                                       pipeline_inputs={{ operation.pipeline_inputs }},
                                                       pipeline_outputs={{ operation.pipeline_outputs }},
+        {% if operation.cpu_request %}
                                                       cpu_request='{{ operation.cpu_request }}',
-                                                      mem_request='{{ operation.mem_request }}G',
+        {% endif %}
+        {% if operation.mem_request %}
+                                                      mem_request='{{ operation.mem_request }}',
+        {% endif %}
+        {% if operation.gpu_request %}
```
Comment (kevin-bates) [inline elyra/templates/kfp_template.jinja2:31]: Should this be `operation.gpu_limit`?
```suggestion
        {% if operation.gpu_limit %}
```

Comment (lresende) [reply elyra/templates/kfp_template.jinja2:31]: @akchinSTC ^^^

Comment (akchinSTC) [reply elyra/templates/kfp_template.jinja2]: good catch, fixed

File: elyra/templates/kfp_template.jinja2 [hunk 2]
```diff
@@ -22,9 +22,15 @@ def create_pipeline():
                                                       cos_dependencies_archive='{{ operation.cos_dependencies_archive }}',
                                                       pipeline_inputs={{ operation.pipeline_inputs }},
                                                       pipeline_outputs={{ operation.pipeline_outputs }},
+        {% if operation.cpu_request %}
                                                       cpu_request='{{ operation.cpu_request }}',
-                                                      mem_request='{{ operation.mem_request }}G',
+        {% endif %}
+        {% if operation.mem_request %}
+                                                      mem_request='{{ operation.mem_request }}',
```
Comment (lresende) [inline elyra/templates/kfp_template.jinja2:29]: do you need the G here for memory?

Comment (lresende) [reply elyra/templates/kfp_template.jinja2:29]: Checked with @akchinSTC and this is already part of the operator

**Review by lresende — APPROVED** [review]

**Review by kevin-bates — APPROVED** [review]
Thanks Alan.

**Review by lresende — APPROVED** [review]

**Pull request merged**